lib/tic_tac_toe_rules.rb

```diff
@@ -0,0 +1,40 @@
+class TicTacToeRules
+    def winner?(player_stats)
+        return false if player_stats.length <= 2 
+        evaluate_player_stats(player_stats)
+    end
+
+    def evaluate_player_stats(player_stats)
+        player_stats = player_stats.sort()
+        match_box = []
+
+        winning_combos.map do |combo|
+            match_box = []
+            player_stats.map do |stat|
+                if combo.include?(stat)
+                    match_box << stat
+                end
+            end
+            return true if match_box.length == 3
+        end
+        false
+    end
+
+    def winning_combos
+        [
+            [0,1,2],
+            [3,4,5],
+            [6,7,8],
+            [0,3,6],
+            [1,4,7],
+            [2,5,8],
+            [0,4,8],
+            [2,4,6]
+        ]
+    end
+
+    def draw?(player_stats)
+        return false unless player_stats.length == 4
+        evaluate_player_stats(player_stats) == false
+    end
+end
```

Comment:
What is `player_stats` supposed to represent in these functions?

Reply:
That's a good question. I'm not entirely sure yet what it will include, but for now I envisioned that the player stats would include an array of marker indexes. So, if I'm player X and I have made two moves on the board, my player stats could be [2,4].

Reply:
Why would that info be relevant in this class? (This is not a trick question)

Reply:
Sorry, didn't see this until now. Player information is not relevant here.